mm/kfence.c

```diff
@@ -463,17 +464,27 @@ static void kfence_dump_object(int obj_index)
 	}
 }
 
+static inline void kfence_start_report(void)
+{
+	console_lock();
```

Comment:
Is this going to have other strange side-effects? Is this approved usage of 'console_lock()'?

At least a comment assuring that this is an ok thing to do would be good.

Reply:
I think so. At least it is in the header, is exported (i.e. available to modules) and has an unambiguous comment: https://elixir.bootlin.com/linux/latest/source/kernel/printk/printk.c#L2278

I'll add a comment to our code to clarify that.

Reply:
What I'm worried about here is the "Acquires a lock which guarantees that the caller has exclusive access to the console system [...]", and then subsequent call to printk. What does "console system" here imply? I think it's the internal state of the console system, but says nothing about whether or not calling printk() still works.

The usage across the kernel I see is console drivers etc.

If we know that 'console_lock()' is always re-entrant, then we can probably make the claim that it's safe. If there is no guarantee that it's re-entrant, then we might eventually have a problem.

Reply:
Turns out it is not re-entrant, at least calling console_lock() twice freezes the task :(